ocaml/testsuite/tests/typing-layouts-products/product_arrays.ml

```diff
@@ -2024,3 +2024,106 @@ Error: This expression has type "#(string * a * bool option) array"
          ([@layout_poly] forces all variables of layout 'any' to be
          representable at call sites).
 |}]
+
+(****************************************************)
```

Comment:
Good tests for literals here. I think we thought it might be helpful to leave a CR to make sure these are also tested in the middle/back end when that becomes possible.